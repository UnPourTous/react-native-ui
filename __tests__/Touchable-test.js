/* eslint-disable no-undef */

import React from 'react'
import {
  Touchable
} from '../src'
import {
  Text,
  View
} from 'react-native'
import renderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'

test('Touchable(Highlight) renders correctly', () => {
  const node = renderer.create(
    <Touchable
      backgroundColor={'red'}
      onPress={() => {
        expect(false).toEqual(true)
      }}
      disabled
      activeColor={'blue'}
      activeType={Touchable.PressMode.highlight}>
      <Text>Highlight Touchable</Text>
    </Touchable>
  )
  ReactTestUtils.Simulate.click(node)

  const tree = node.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Touchable(Opacity) renders correctly', () => {
  const tree = renderer.create(
    <Touchable
      backgroundColor={'red'}
      onPress={() => {}}
      disabled
      activeType={Touchable.PressMode.opacity}>
      <Text>Opacity Touchable</Text>
    </Touchable>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Touchable(Mask) renders correctly', () => {
  const tree = renderer.create(
    <View style={{flex: 1}}>
      <Touchable
        backgroundColor={'red'}
        onPress={() => {}}
        disabled
        activeType={Touchable.PressMode.mask} title={'Test Button'}>
        <Text>Mask Touchable</Text>
      </Touchable>
    </View>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Touchable(None) renders correctly', () => {
  const tree = renderer.create(
    <View style={{flex: 1}}>
      <Touchable
        backgroundColor={'red'}
        onPress={() => {}}
        disabled
        activeType={Touchable.PressMode.none} title={'Test Button'}>
        <Text>None Touchable</Text>
      </Touchable>
    </View>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
