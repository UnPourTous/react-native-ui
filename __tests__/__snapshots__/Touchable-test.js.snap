// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable(Highlight) renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 0.3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Highlight Touchable
    </Text>
  </View>
</View>
`;

exports[`Touchable(Mask) renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "opacity": 0.3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Mask Touchable
      </Text>
    </View>
  </View>
</View>
`;

exports[`Touchable(None) renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "opacity": 0.3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        None Touchable
      </Text>
    </View>
  </View>
</View>
`;

exports[`Touchable(Opacity) renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 0.3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Opacity Touchable
    </Text>
  </View>
</View>
`;
